### ISSUE TITLE
wrong ELF class: ELFCLASS32

### ISSUE THREAD

**Issue by beduffy** [novelda/Legacy-SW#4]
I know many of the devices have been discontinued but it would be great to know how to make the pymoduleconnector library work on Jetson Nano. 
![image](https://user-images.githubusercontent.com/9202993/113833021-77229180-9789-11eb-9aa9-96ff5557d1a7.png)

**Comment by root27** [comment]
Could you hrlp me to find python api for module connector? Because the links that Novelda companu has given are not working.

**Comment by beduffy** [comment]
All I really had was this code
https://github.com/novelda/Legacy-SW/tree/master/ModuleConnector/ModuleConnector-rpi-1/examples

and this code which really really helped me below. I ended up swapping to the SLMX4 from sensorlogic which has a slightly different api. I had to build my own python serial interface to it.
https://github.com/XPengZhao/Respiratory-monitoring

**Comment by root27** [comment]
Thank you . I have solved the problem

**Comment by sudo-bcli** [comment]
@beduffy Hi I just ran into the same problem. I think it's because their dynamic library was compiled for older version of Raspbian (armhf 32bit) but nano is arm64 (64bit).   Since there's no source code of the Module Connector, we can't rebuild a arm64 library on Nano by ourselves.

There's a thread on [Nvidia Forum](https://forums.developer.nvidia.com/t/run-a-32bit-application-on-the-jetson-nano/127050) discussing "How to use arm32 library on arm64".  May worth a shot.

BTW I don't think Novelda would provide any support for this product (note: there's literally a villain on their [home page](https://novelda.com/)(scroll down)), may be [Sensor Logic](https://www.sensorlogic.ai/xethru/) will?

---
### Update Oct 2021 ###

1. SensorLogic replied they "don't make this radar and don't provide any changes".
2. Have to switch from Nano to Raspberry Pi
3. Module Connector (python35-arm-linux-gnueabihf) works on Raspberry Pi 4B with Raspberry OS (Release date: May 7th 2021)

**Comment by G-Bob** [comment]
> @beduffy Hi I just ran into the same problem. I think it's because their dynamic library was compiled for older version of Raspbian (armhf 32bit) but nano is arm64 (64bit). Since there's no source code of the Module Connector, we can't rebuild a arm64 library on Nano by ourselves.
> 
> There's a thread on [Nvidia Forum](https://forums.developer.nvidia.com/t/run-a-32bit-application-on-the-jetson-nano/127050) discussing "How to use arm32 library on arm64". May worth a shot.
> 
> BTW I don't think Novelda would provide any support for this product (note: there's literally a villain on their [home page](https://novelda.com/)(scroll down)), may be [Sensor Logic](https://www.sensorlogic.ai/xethru/) will?
> 
> ### Update Oct 2021
> 1. SensorLogic replied they "don't make this radar and don't provide any changes".
> 2. Have to switch from Nano to Raspberry Pi
> 3. Module Connector (python35-arm-linux-gnueabihf) works on Raspberry Pi 4B with Raspberry OS (Release date: May 7th 2021)

@sudo-bcli Hi, I tried the specific Raspbian OS image. But still got the error of libboost-dev version mismatch: 

ImportError: libboost_filesystem.so.1.62.0: cannot open shared object file: No such file or directory

The system's libboost version is 1.67.0. Meanwhile I cannot find any source of the 1.62.0. Can you share the experience how you connect with Raspberry PI4? 

Many thanks.

**Comment by G-Bob** [comment]
> All I really had was this code https://github.com/novelda/Legacy-SW/tree/master/ModuleConnector/ModuleConnector-rpi-1/examples
> 
> and this code which really really helped me below. I ended up swapping to the SLMX4 from sensorlogic which has a slightly different api. I had to build my own python serial interface to it. https://github.com/XPengZhao/Respiratory-monitoring

@beduffy Hi, many thanks for your sharing. Do you know if this radar support arm based devices?

**Comment by beduffy** [comment]
Yep @G-Bob  SLMX4 works on our Jetson Nano and Xavier and those are ARM based.

**Comment by G-Bob** [comment]
> Yep @G-Bob SLMX4 works on our Jetson Nano and Xavier and those are ARM based.

Thank you @beduffy, just wonder if there any link for driver based on ARM. Because in their project, there are only python demo based on windows and linux: https://github.com/SensorLogicInc/modules/tree/main/python, but not mention if it's ARM based or AMD.
Have you tested the code of this page on the edge devices? 

Sincerely thanks for your reply.

**Comment by aabsz** [comment]
Hi everyone.
Does anyone know how to fix the "wrong ELF class: ELFCLASS32" error with the X4M300 (X4M03) sensor?

And one more question. I need the raw data from UWB. I see some of you are using SLMX4 from sensor logic. Do you know if this sensor, like X4M03, provides us with its raw data?

**Comment by sudo-bcli** [comment]
> > @beduffy Hi I just ran into the same problem. I think it's because their dynamic library was compiled for older version of Raspbian (armhf 32bit) but nano is arm64 (64bit). Since there's no source code of the Module Connector, we can't rebuild a arm64 library on Nano by ourselves.
> > There's a thread on [Nvidia Forum](https://forums.developer.nvidia.com/t/run-a-32bit-application-on-the-jetson-nano/127050) discussing "How to use arm32 library on arm64". May worth a shot.
> > BTW I don't think Novelda would provide any support for this product (note: there's literally a villain on their [home page](https://novelda.com/)(scroll down)), may be [Sensor Logic](https://www.sensorlogic.ai/xethru/) will?
> > ### Update Oct 2021
> > 
> > 1. SensorLogic replied they "don't make this radar and don't provide any changes".
> > 2. Have to switch from Nano to Raspberry Pi
> > 3. Module Connector (python35-arm-linux-gnueabihf) works on Raspberry Pi 4B with Raspberry OS (Release date: May 7th 2021)
> 
> @sudo-bcli Hi, I tried the specific Raspbian OS image. But still got the error of libboost-dev version mismatch:
> 
> ImportError: libboost_filesystem.so.1.62.0: cannot open shared object file: No such file or directory
> 
> The system's libboost version is 1.67.0. Meanwhile I cannot find any source of the 1.62.0. Can you share the experience how you connect with Raspberry PI4?
> 
> Many thanks.

Really sorry that I didn't see your message earlier, guess it was submerged in tons of other useless Github notifications. 
Currently I'm reworking on that old project back in 2021, please let me know if you still need any help with getting X4M300 working with Raspberry Pi.

**Comment by aabsz** [comment]
Hi, 
No problem. Thank you for your reply. 
I recently borrowed an SLMX4 from sensorlogic and I have access to all that I need. However, I still have a problem with setting up X4M300. Do you have any tips on how to set it up in Linux? (not Raspberry Pi)

**Comment by sudo-bcli** [comment]
> Hi, No problem. Thank you for your reply. I recently borrowed an SLMX4 from sensorlogic and I have access to all that I need. However, I still have a problem with setting up X4M300. Do you have any tips on how to set it up in Linux? (not Raspberry Pi)

Unfortunately, no, I've only tried X4M300 with Raspberry Pi and and Jetson Nano.

**Comment by aabsz** [comment]
> > Hi, No problem. Thank you for your reply. I recently borrowed an SLMX4 from sensorlogic and I have access to all that I need. However, I still have a problem with setting up X4M300. Do you have any tips on how to set it up in Linux? (not Raspberry Pi)
> 
> Unfortunately, no, I've only tried X4M300 with Raspberry Pi and and Jetson Nano.

OK. I will try Raspberryby later then.

**Comment by shibingli** [comment]
> Thank you . I have solved the problem

May I ask how did you solve it? I have several X4M03 development kits here, and due to some reasons, I have to use this device. However, I am unable to use it based on the current Python library.
